### PR TITLE
MULE-10134: Migrate extensions to use isolated test runner - Integration tests

### PR DESCRIPTION
**Pull request by gsfernandes**

### CODE REVIEW

File: tests/functional/src/main/java/org/mule/functional/functional/FunctionalTestNotification.java [hunk 1]
```diff
@@ -40,7 +40,15 @@
 
     static
     {
-        registerAction("event received", EVENT_RECEIVED);
+        //TODO(pablo.kraan): find a better way to manage this situation.
```
Comment (pablokraan) [inline tests/functional/src/main/java/org/mule/functional/functional/FunctionalTestNotification.java]: put your name here

Comment (pablokraan) [inline tests/functional/src/main/java/org/mule/functional/functional/FunctionalTestNotification.java]: This change was required when the classloader was created per test. Are you sure that is still required?

File: tests/integration-tranports/src/test/java/org/mule/compatibility/config/spring/parsers/AbstractBadConfigTestCase.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+import org.mule.runtime.core.context.DefaultMuleContextFactory;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A stripped-down version of FunctionalTestCase that allows us to test the parsing of a bad configuration. 
+ */
+//TODO(gfernandes): remove this class and use the one from it tests once transport it test are migrated to use isolation runner
+public abstract class AbstractBadConfigTestCase extends FunctionalTestCase
```
Comment (pablokraan) [inline tests/integration-tranports/src/test/java/org/mule/compatibility/config/spring/parsers/AbstractBadConfigTestCase.java]: it tests > IT tests?

File: tests/integration-tranports/src/test/java/org/mule/compatibility/config/spring/parsers/AbstractBadConfigTestCase.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+import org.mule.runtime.core.api.config.ConfigurationBuilder;
+import org.mule.runtime.core.context.DefaultMuleContextFactory;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A stripped-down version of FunctionalTestCase that allows us to test the parsing of a bad configuration. 
+ */
+//TODO(gfernandes): remove this class and use the one from it tests once transport it test are migrated to use isolation runner
```
Comment (pablokraan) [inline tests/integration-tranports/src/test/java/org/mule/compatibility/config/spring/parsers/AbstractBadConfigTestCase.java]: Add a reference to a Jira issue in each TODO item

File: tests/integration-tranports/src/test/java/org/mule/test/integration/message/AbstractPropertyScopeTestCase.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+import org.mule.tck.junit4.rule.DynamicPort;
+
+import org.junit.Rule;
+import org.junit.Test;
+
+//TODO(gfernandes): remove this class and use the one from it tests once transport it test are migrated to use isolation runner
+public abstract class AbstractPropertyScopeTestCase extends FunctionalTestCase
+{
+    @Rule
+    public DynamicPort port1 = new DynamicPort("port1");
```
Comment (pablokraan) [inline tests/integration-tranports/src/test/java/org/mule/test/integration/message/AbstractPropertyScopeTestCase.java]: There is only one port. port1 > port

File: tests/integration/pom.xml [hunk 5]
```diff
@@ -12,20 +12,30 @@
 	<description>Tests which ensure that Mule sub projects work well together.</description>
 	<packaging>jar</packaging>
 
-    <properties>
-        <skipExportTests>false</skipExportTests>
-        <skipTests>${skipIntegrationTests}</skipTests>
-    </properties>
+	<properties>
```
Comment (pablokraan) [inline tests/integration/pom.xml]: formatting of this XML file does not seems correct. We use 4 chars indentation, not 8.

File: tests/integration/src/test/java/org/mule/test/AbstractIntegrationTestCase.java [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+
+package org.mule.test;
+
+import org.mule.functional.junit4.MuleArtifactFunctionalTestCase;
+
+public abstract class AbstractIntegrationTestCase extends MuleArtifactFunctionalTestCase
```
Comment (pablokraan) [inline tests/integration/src/test/java/org/mule/test/AbstractIntegrationTestCase.java]: Add javadoc describing the purpose of this class

File: tests/integration/src/test/java/org/mule/test/construct/FlowConfigurationFunctionalTestCase.java [hunk 7]
```diff
@@ -137,6 +137,7 @@ public void testInOutAppendFlow() throws Exception
     }
 
     @Test
+    @Ignore("ArtifactClassLoaderRunner: groovy issue")
```
Comment (pablokraan) [inline tests/integration/src/test/java/org/mule/test/construct/FlowConfigurationFunctionalTestCase.java]: Add Jira reference for each ignored test

File: tests/integration/src/test/resources/org/mule/test/el/expression-language-config.xml [hunk 8]
```diff
@@ -24,7 +24,7 @@
 
     <flow name="flow">
         <expression-component>regexWrap('hello')</expression-component>
-        <test:assert expression="#[payload == 'foo']" />
+        <validation:is-true expression="#[payload == 'foo']"/>
```
Comment (pablokraan) [inline tests/integration/src/test/resources/org/mule/test/el/expression-language-config.xml]: Why is this change required?

Comment (gsfernandes) [inline tests/integration/src/test/resources/org/mule/test/el/expression-language-config.xml]: In order to use validation extension, no other place or test is using it (from a config at least). The reason was to have an integration test that uses the validation extension on isolated test mode.

File: tests/functional/src/main/java/org/mule/functional/functional/FunctionalTestNotification.java (hunk 1 above)
Comment (gsfernandes) [inline tests/functional/src/main/java/org/mule/functional/functional/FunctionalTestNotification.java]: I'll revert this and run all the test to check if this is really necessary. I will let you know, meanwhile I'm looking this PR.

**Comment by gsfernandes** [comment]
@pablokraan I just confirmed that test passed with the suggested changes in this PR. Feel free to continue the review. Thank you!
